execution/aggregate/khashaggregate.go

```diff
@@ -205,41 +211,112 @@ func (a *kAggregate) init(ctx context.Context) error {
 	return nil
 }
 
-func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64) {
-	for i, sId := range sampleIDs {
-		h := a.inputToHeap[sId]
-		switch {
-		case h.Len() < k:
-			heap.Push(h, &entry{sId: sId, total: samples[i]})
+func (a *kAggregate) aggregate(t int64, result *[]model.StepVector, k int, sampleIDs []uint64, samples []float64, histogramIDs []uint64, histograms []*histogram.FloatHistogram) {
+	groupsRemaining := len(a.heaps)
 
-		case h.compare(h.entries[0].total, samples[i]) || (math.IsNaN(h.entries[0].total) && !math.IsNaN(samples[i])):
-			h.entries[0].sId = sId
-			h.entries[0].total = samples[i]
+	if a.aggregation == parser.TOPK || a.aggregation == parser.BOTTOMK {
+		for i, sId := range sampleIDs {
+			h := a.inputToHeap[sId]
```

Comment:
Maybe small nit, but with histogram code here, `h` is maybe not the greatesy var name. I wonder if we can just name it sampleHeap?